sanic/blueprints.py

```diff
@@ -208,3 +237,24 @@ def register(self, app, options):
         self.middlewares = middleware
         self.exceptions = exception_handlers
         self.listeners = dict(listeners)
+
+    async def dispatch(self, *args, **kwargs):
+        where = kwargs.pop("where", {})
```

Comment:
Makes sense. The name sort of stuck from the underlying implementation that is meant to be more generalized. I agree we could do a better job here with variable names.

Reply:
:thinking: They are both set to `where` right now. 

Let me first backup and explain. The `requirements` attribute on the `signal` is akin to what the router is using under the hood. Initially, its purpose is to help facilitate virtual host routing:

```python
app.get("/", host="example.com")()
```

At somepoint, I would like to expand its usage to also include routing based upon cookies, headers, etc. But there also might be a need for requirements to hold more structured meta data. I do not want to get too far ahead, but I also want to allow for flexibility to grow. Hence, it is a dictionary.

For now, we can simply pass through that whole object. At some point, maybe we need to add some logic.

In the current usage in signals, it adds the name of the blueprint to the object both with `signal` and `dispatch`.

So, `requirements` is a vestige of the underlying API. I agree that we can expose another term that makes more sense. However, whatever we choose I want to make sure is clear and unambiguous enough that someone can easily pick up the code and let it be obvious what is happening.

---

And this is where I am torn. Originally my thought was that they needed to be two terms since the definition on the signal would potentially carry more details and meta information. On the `dispatch` it is just passing matching criteria.

On the other hand, having two terms can ultimately be confusing. Especially if they are both beginning with `w`. I like `where` because it is a familiar construct in query languages. I think `matches` and `provides` would be the types of keywords that someone would forget and need to look up every time they want to use it. _Was it `match` or `matches`? Does that go in `signal` or `dispatch`?_

I definitely am not a fan of a verb here. I still think `where` is easiest to remember. And, while I originally like having two names for future flexibility, I do not want to make a design decision on a feature that may or may not come into existence in the future.

---

I guess that was a long way to say that I am in favor of a single argument still. If I had to rank the proposals:

1. where
1. condition
1. when
1. extra
1. with

- ~needs~ 
- ~provides`
- ~matches`
- ~arguments~ (confusing what this means, especially since we have another keyword that will provide arguments to the handler

Reply:
@sanic-org/core-devs What if we go with `condition` for both right now:

```python
@app.signal("foo.bar.baz", condition={"one": "two"})
await app.dispatch("foo.bar.baz", condition={"one": "two"})
```

Then, _IF_ in the future we want to add more logic to the matching, we can add another kwarg:

```python
await app.dispatch("foo.bar.baz", match={
    "conditions": [{"one": "two"}, {"one": "three"}],
    "rules": RouteRules.MATCH_ANY
}
```